core/src/tangram.cpp

```diff
@@ -98,11 +66,27 @@ void initialize(const char* _scenePath) {
     // Label setup
     m_labels = std::make_unique<Labels>();
 
+    // Create an AsyncWorker
+    m_asyncWorker = std::make_unique<AsyncWorker>();
+
     LOG("finish initialize");
 
 }
 
-void setScene(std::shared_ptr<Scene>& _scene) {
+Map::~Map() {
+    // Explicitly destroy all member objects so that we have a chance
+    // to run any resulting jobs sent to the JobQueue.
+    m_asyncWorker.reset();
+    m_inputHandler.reset();
+    m_labels.reset();
+    m_view.reset();
+    m_nextScene.reset();
+    m_tileManager.reset();
+    m_tileWorker.reset();
+    JobQueue::runJobsForCurrentThread();
```

Comment:
Ha! looks like I was handling this already from async scene loading code on an app exit! which is same as Map object Destruction